eng/pipelines/tools-repo-versioning.yml

```diff
@@ -0,0 +1,36 @@
+# Tag the tools repos (azure-sdk-tools or azure-sdk-build-tools) to create some form of versioning.
+resources:
+  repositories:
+    - repository: azure-sdk-tools
+      type: github
+      name: Azure/azure-sdk-tools
+      endpoint: azure
+    - repository: azure-sdk-build-tools
+      type: git
+      name: internal/azure-sdk-build-tools
+
+trigger: none
+
+pr: none
+
+pool:
+  vmImage: windows-2019
+
+jobs:
+  - job: TagToolsRepo
+    displayName: Version Repo with Git Tag
+
+    steps:
+    - checkout: self
```

Comment:
Why do you need to checkout both? I would expect you only need to check out the one, especially if you make the steps inline.

Reply:
So I wanted to make this work on 'azure-sdk-tools' as well as 'azure-sdk-build-tools'. The git command needs to be run on the repo requiring versioning. So I think the appriopriate repo needs to be checked out. Right?

Reply:
Wouldn't that be handled by us having 2 pipelines? this would run in the context of the repo that we set it up for.

Reply:
Even  with two pipelines we will still need to checkout the appropriate Repo. Since the pipeline definition will be resident in `azure-sdk-tools` that will be checkout by default. For `azure-sdk-build-tools` there needs to be a way to specify that it should be checked out otherwise the git commands will not work. Except the yml definition is also present in `azure-sdk-build-tools`. I think its still best to use one script and set the repos to be checked out and the PAT using pipeline variables then condition it on the script.